Tests/OpenCombineTests/PublisherTests/EncodeTests.swift

```diff
@@ -0,0 +1,100 @@
+//
+//  EncodeTests.swift
+//
+//
+//  Created by Joseph Spadafora on 6/21/19.
+//
+
+import XCTest
+
+#if OPENCOMBINE_COMPATIBILITY_TEST
+import Combine
+#else
+import OpenCombine
+#endif
+
+@available(macOS 10.15, *)
+final class EncodeTests: XCTestCase {
+    static let allTests = [
+        ("testEncodeWorks", testEncodeWorks),
+        ("testDemand", testDemand),
+        ("testEncodeSuccessHistory", testEncodeSuccessHistory)
+    ]
+
+    private var encoder = TestEncoder()
+    private var decoder = TestDecoder()
+
+    override func setUp() {
+        encoder = TestEncoder()
+        decoder = TestDecoder()
+    }
+
+    func testEncodeWorks() throws {
+        // Given
+        let testValue = ["test": "TestDecodable"]
+        let subject = PassthroughSubject<[String: String], Error>()
+        let publisher = subject.encode(encoder: encoder)
+        let subscriber = TrackingSubscriberBase<Int, Error>(
+            receiveSubscription: { $0.request(.unlimited) }
+        )
+
+        // When
+        publisher.subscribe(subscriber)
+        subject.send(testValue)
+
+        // Then
+        XCTAssert(encoder.encoded.first?.value as? [String: String] == testValue)
+    }
+
+    func testEncodeSuccessHistory() throws {
+        // Given
+        let testValue = ["test": "TestDecodable"]
+        let subject = PassthroughSubject<[String: String], Error>()
+        let publisher = subject.encode(encoder: encoder)
+        let subscriber = TrackingSubscriberBase<Int, Error>(
+            receiveSubscription: { $0.request(.unlimited) }
+        )
+
+        // When
+        publisher.subscribe(subscriber)
+        subject.send(testValue)
+
+        // Then
+        guard let testKey = encoder.encoded.first?.key, encoder.encoded.count == 1 else {
+            throw "Could not get testing data from encoding" as TestingError
```

Comment:
Can we use `XCTFail` here instead of throwing? In that case XCTest will show a line number where the error occurred.